pkg/cfn/builder/vpc_ipv6.go

```diff
@@ -35,23 +36,58 @@ func (v *IPv6VPCResourceSet) CreateTemplate() (*gfnt.Value, *SubnetDetails, erro
 		EnableDnsSupport:   gfnt.True(),
 		EnableDnsHostnames: gfnt.True(),
 	})
+	v.rs.defineOutput(outputs.ClusterVPC, vpcResourceRef, true, func(val string) error {
+		v.clusterConfig.VPC.ID = val
+		return nil
+	})
 
 	v.rs.newResource(IPv6CIDRBlockKey, &gfnec2.VPCCidrBlock{
 		AmazonProvidedIpv6CidrBlock: gfnt.True(),
 		VpcId:                       gfnt.MakeRef(VPCResourceKey),
 	})
+	addSubnetOutput := func(subnetRefs []*gfnt.Value, topology api.SubnetTopology, outputName string) {
+		v.rs.defineJoinedOutput(outputName, subnetRefs, true, func(value string) error {
+			return vpc.ImportSubnetsFromIDList(v.ec2API, v.clusterConfig, topology, strings.Split(value, ","))
+		})
+	}
+
+	var privateSubnets []SubnetResource
+	cidrPartitions := (len(v.clusterConfig.AvailabilityZones) * 2) + 2
+	for i, az := range v.clusterConfig.AvailabilityZones {
+		azFormatted := formatAZ(az)
+		rtRef := v.rs.newResource(PrivateRouteTableKey+azFormatted, &gfnec2.RouteTable{
+			VpcId: gfnt.MakeRef(VPCResourceKey),
+		})
+
+		subnet := v.createSubnet(az, azFormatted, i+len(v.clusterConfig.AvailabilityZones), cidrPartitions, true)
+		privateSubnetResourceRefs = append(privateSubnetResourceRefs, subnet)
+		privateSubnets = append(privateSubnets, SubnetResource{
+			Subnet:           subnet,
+			AvailabilityZone: az,
+			RouteTable:       rtRef,
+		})
+
+		v.rs.newResource(PrivateRouteTableAssociation+azFormatted, &gfnec2.SubnetRouteTableAssociation{
+			RouteTableId: rtRef,
+			SubnetId:     subnet,
+		})
+	}
+	addSubnetOutput(privateSubnetResourceRefs, api.SubnetTopologyPrivate, outputs.ClusterSubnetsPrivate)
 
+	if v.isFullyPrivate() {
+		return vpcResourceRef, &SubnetDetails{
+			Private: privateSubnets,
+		}, nil
+	}
```

Comment:
Technically, the moving around is what mattered, but I also added things to the subnet such as routing table id and the AZ which was missing, and removed some of the route creation since those were not required in private mode and that's about it I think.